providers/azureblob/src/test/java/org/jclouds/azureblob/config/AzureBlobHttpApiModuleTest.java

```diff
@@ -0,0 +1,94 @@
+public class AzureBlobHttpApiModuleTest {
+
+   AzureBlobHttpApiModule module = new AzureBlobHttpApiModule();
+   String identity = "identity";
+   
+   @Test 
+   void testAuthSasNonSufficientParametersSvSe(){
+      Credentials creds = new Credentials(identity, "sv=2018-03-28&se=2019-02-14T11:12:13Z");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasNonSufficientParametersSvSeSp(){
+      Credentials creds = new Credentials(identity, "sv=2018-03-28&se=2019-02-14T11:12:13Z&sp=abc&st=2019-01-20T11:12:13Z");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasSharedKey(){
+      Credentials creds = new Credentials(identity, "u2iAP01ARTewyK/MhOM1d1ASPpjqclkldsdkljfas2kfjkh895ssfslkjpXKfhg==");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasRandomString(){
+      Credentials creds = new Credentials(identity, "sadf;gjkhflgjkhfdlkfdljghskldjghlfdghw4986754ltjkghdlfkjghst;lyho56[09y7poinh");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasRandomQueryString(){
+      Credentials creds = new Credentials(identity, "a=apple&b=banana&c=cucumber&d=diet");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasRandomQueryString2(){
+      Credentials creds = new Credentials(identity, "sva=swajak&sta=stancyja&spa=spakoj&sea=mora&sig=podpis");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
+   
+   @Test 
+   void testAuthSasAccountSAS(){
+      Credentials creds = new Credentials(identity, "sv=2018-03-28&ss=b&srt=sco&sp=r&se=2019-02-13T17:03:09Z&st=2019-02-13T09:03:09Z&spr=https&sig=wNkWK%2GURTjHWhtqG6Q2Gu%2Qu%3FPukW6N4%2FIH4Mr%2F%2FO42M%3D");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), true);
+   }
+   
+   @Test 
+   void testAuthSasServiceSAS(){
+      Credentials creds = new Credentials(identity, "sp=rl&st=2019-02-14T08:50:26Z&se=2019-02-15T08:50:26Z&sv=2018-03-28&sig=Ukow8%2GtpQpAiVZBLcWp1%2RSpFq928MAqzp%2BdrdregaB6%3D&sr=b");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), true);
+   }
+   
+   @Test 
+   void testAuthSasEmptyString(){
+      Credentials creds = new Credentials(identity, "");
+      Supplier<Credentials> supplier = Suppliers.ofInstance(creds);
+      assertEquals(module.authSAS(supplier), false);
+   }
```

Comment:
This class has a lot of repeated code. Put all the tokens in an array and better use the TestNG data provider to run all tests:

```java
@DataProvider(name = "sas-tokens")
public static Object[][] tokens() {
   return new Object[][] {
      {false, ""},
      {true, "sp=rl&st=2019-02-14T08:50:26Z&se=2019-02-15T08:50:26Z&sv=2018-03-28&sig=Ukow8%2GtpQpAiVZBLcWp1%2RSpFq928MAqzp%2BdrdregaB6%3D&sr=b"},
      ...
   };
}

@Test(dataProvider = "sas-tokens")
private void testSASAuthentication(boolean expected, String credential) {
   AzureBlobHttpApiModule module = new AzureBlobHttpApiModule();
   Credentials creds = new Credentials("identity", credential);
   assertEquals(module.authSAS(Suppliers.ofInstance(creds)), expected);
}
```

Reply:
Thanks a lot for your suggestions, I have changed it.